pkgs/top-level/all-packages.nix

```diff
@@ -150,6 +150,12 @@ with pkgs;
 
   appindicator-sharp = callPackage ../development/libraries/appindicator-sharp { };
 
+  canonicalize-jar = callPackage ../build-support/java/canonicalize-jar.nix { };
+  canonicalize-jars-hook = makeSetupHook {
```

Comment:
Maybe the commit that adds these hook and script should use them in `pkgs/build-support/release/functions.sh` somehow? I dislike that we have such code duplication.

Reply:
I disliked it too, but the functions they're duplicated from are removed next commit, and it seemed like busywork.

These commits were originally merged, but splitting them up allowed the commit messages to get clearer.